src/Tizen.NUI.Scene3D/src/public/ModelMotion/MotionData.cs

```diff
@@ -16,109 +16,384 @@
+    /// keyFrames.Add(0.0f, 0.0f);
+    /// keyFrames.Add(0.0f, 1.0f);
+    /// motionData.Add(new MotionTransformIndex(new PropertyKey("nodeName"), MotionTransformIndex.TransformType.PositionX), new MotionValue(keyFrames));
+    ///
+    /// // Make MotionIndex with BlendShapeIndex
+    /// motionData.Add(new BlendShapeIndex(new PropertyKey("nodeName"), new PropertyKey("blendShapeName")), motionData.GetValue(1u));
+    /// </code>
+    ///
+    /// We can request to load MotionData from file or buffer asynchronously.
+    /// If load completed, <see cref="LoadCompleted"/> event will be invoked.
+    /// If we try to load before LoadCompleted event invoked, previous load request cancel and only latest request loaded.
+    ///
+    /// <code>
+    /// MotionData motionData = new MotionData();
+    /// motionData.LoadCompleted += OnLoadCompleted;
+    /// motionData.LoadBvh("bvhFilename.bvh", Vector3.One);
+    ///
+    /// ...
+    ///
+    /// void OnLoadCompleted(object o, event e)
+    /// {
+    ///     MotionData motionData = o as MotionData;
+    ///     /// Do something.
+    /// }
+    /// </code>
+    ///
+    /// We can generate animation of Scene3D.Model from MotionData class.
+    /// Or, just set values.
+    ///
+    /// <code>
+    /// // Generate animation from loaded Model
+    /// Animation animation = model.GenerateMotionDataAnimation(motionData);
+    /// animation.Play();
+    ///
+    /// // Set values from loaded Model.
+    /// model2.SetMotionData(motionData);
+    /// </code>
     /// </summary>
     /// <remark>
+    /// We don't check duplicated MotionIndex internally.
     /// We don't check MotionValue type is matched with MotionIndex.
     /// </remark>
     [EditorBrowsable(EditorBrowsableState.Never)]
-    public class MotionData : IDisposable
+    public class MotionData : BaseHandle
     {
-        private List<(MotionIndex, MotionValue)> motionValues { get; set; } = null;
+        private EventHandler loadCompletedEventHandler;
+        private LoadCompletedCallbackType loadCompletedCallback;
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate void LoadCompletedCallbackType(IntPtr motionData);
 
         /// <summary>
-        /// Owned motion value list.
+        /// Create an initialized motion data.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public MotionData() : this(Interop.MotionData.MotionDataNew(), true)
+        {
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+
+        /// <summary>
+        /// Create an initialized motion data with duration.
+        /// </summary>
+        /// <param name="durationSeconds">Duration of this motion data when it be generated as Animation.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public MotionData(float durationSeconds) : this(Interop.MotionData.MotionDataNew(durationSeconds), true)
```

Comment:
NUI handles animation duration with integer value denotes milliseconds.
Please changes float duration to integer.

Reply:
Oh... That was good point!
Match the experience with Tizen.NUI.Animation is important. Fix done.